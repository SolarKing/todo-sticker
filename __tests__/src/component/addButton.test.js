// @flow
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import AddButton from '../../../src/component/addButton';
import AddModal from '../../../src/container/addTodoModalContainer';

describe('AddButton', () => {
    let wrapper: ShallowWrapper;

    beforeAll(() => {
        global.Date.prototype.getTime = jest.fn().mockReturnValue(0);
        wrapper = shallow(<AddButton/>);
    });

    it('should render correct', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should change color when hover', () => {
        expect(wrapper.find('AddButtonIcon')).toHaveStyleRule('color', '#6494c4',{
            modifier: ':hover',
        });
    });

    it('should show add modal and setting time', () => {
        wrapper.find('AddButtonIcon').simulate('click');
        expect(wrapper.state().showEditModal).toBeTruthy();
        expect(wrapper.state().defaultTimestamp).toEqual(0);
    });

    it('should hide add modal when it save triggered', () => {
        wrapper.find(AddModal).at(0).simulate('save');
        expect(wrapper.state().showEditModal).toBeFalsy();
    });

    it('should hide add modal when it cancel triggered', () => {
        wrapper.find(AddModal).at(0).simulate('save');
        expect(wrapper.state().showEditModal).toBeFalsy();
    });

});