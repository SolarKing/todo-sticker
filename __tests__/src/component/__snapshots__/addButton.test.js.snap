// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddButton should render correct 1`] = `
<div>
  <AddButtonIcon
    onClick={[Function]}
  >
    <span
      className="glyphicon glyphicon-plus-sign"
    />
  </AddButtonIcon>
  <Connect(EditModal)
    defaultDescription=""
    defaultDueDatetime={0}
    onCancel={[Function]}
    onSave={[Function]}
    title="Add Event"
    visible={false}
  />
</div>
`;
